platform/testing/end-to-end/cypress-upgrade-guide.md

```diff
@@ -0,0 +1,182 @@
+
+The upgrade from Cypress 4.8 to Cypress 6.2.1 introduces several changes to test structure, syntax, and other aspects of Cypress tests. This guide covers key changes that are most likely to affect existing tests. There are a handful of deprecated methods that should be updated before the core Cypress library no longer supports them. A full, detailed guide of all changes can be found in the [additional resources](#additional-resources) section.
+
+## Migration changes <a name="migration-changes"></a>
+
+### `cy.intercept` vs. `cy.route` <a name="intercept"></a>
+
+[`cy.server()`](https://docs.cypress.io/api/commands/server.html) and [`cy.route()`](https://docs.cypress.io/api/commands/route.html) are being deprecated in favor of [`cy.intercept()`](https://docs.cypress.io/api/commands/intercept.html). `cy.server()` is not needed when using `cy.intercept()`. Syntax for `cy.intercept()` is the same as it is for `cy.route()`.
+
+For example, this block of code:
+
+```javascript
+cy.server();
+cy.route('/hello').as('helloWorld');
+```
+
+can be re-written as
+
+```javascript
+cy.intercept('/hello').as('helloWorld');
+```
+
+`cy.wait()` also behaves differently with `cy.intercept()`:
+
+```javascript
+cy.server();
+cy.route('/hello').as('helloWorld');
+cy.wait('@helloWorld')
+  .then(({ requestBody, responseBody, status }) => {
+    expect(status).to.eq(200);
+  });
+```
+
+now looks like:
+
+```javascript
+cy.intercept('/hello').as('helloWorld');
+cy.wait('@helloWorld')
+  .then(({ request, response }) => {
+    expect(response.statusCode).to.eq(200);
+  });
+```
+
+### Fixtures <a name="fixtures"></a>
+
+`cy.intercept()` includes a `routeHandler` argument in which fixture paths can be specified. The syntax is slightly different than with `cy.route()`:
+
+```javascript
+cy.server();
+cy.route('GET', '/hello', 'fx:helloWorld');
+```
+
+now looks like:
+
+```javascript
+cy.intercept('GET', '/hello', {
+  fixture: 'helloWorld'
+});
+```
+
+### Non-existent elements <a name="non-existent-elements"></a>
+
+Cypress 6 introduces changes that affect how assertions behave for non-existent elements. Cases that used to pass now fail, as Cypress 6 is more strict about assertions for these kinds of elements.
+
+Examples of assertions that used to pass but now fail:
+
+```javascript
+cy.get('#does-not-exist').should('not.be.visible');
+cy.get('#does-not-exist').should('not.have.class', 'present');
+cy.get('#does-not-exist').should('not.contain', 'text');
+```
+
+With Cypress 6, the preferred approach is to simply check that the element does not exist:
+
+```javascript
+cy.get('#does-not-exist').should('not.exist');
+```
+
+### Opacity of elements <a name="opacity"></a>
+
+Cypress 6 treats elements with the CSS property `opacity: 0` as not visible, but these elements can still be interacted with. In older versions of Cypress the following assertion would fail, but now passes:
+
+```javascript
+// The `.zero-opacity` element has the CSS property `opacity: 0`.
```

Comment:
`zero-opacity` is/would be a CSS utility class like [Tailwind's `opacity-0`](https://tailwindcss.com/docs/opacity), correct?

Reply:
I gave it that name to make the example more clear, but yeah, similar to your link.